File: backend-ruby/app/models/sinh_vien.rb
class SinhVien < ApplicationRecord
  self.table_name = 'SinhVien'
  self.primary_key = 'SinhVienID'
  belongs_to :nguoi_dung, class_name: "NguoiDung", foreign_key: "NguoiDungID", optional: true
  has_many :dang_ky_thi_records, class_name: "DangKyThi", foreign_key: "SinhVienID"
end

File: backend-csharp/Models/SinhVien.cs
using System.Text.Json.Serialization;

namespace backend_csharp.Models;

public class SinhVien
{
    public int SinhVienID { get; set; }
    public string MaSinhVien { get; set; } = string.Empty;
    public string HoTen { get; set; } = string.Empty;
    public string? Lop { get; set; }
    public string? Email { get; set; }
    public string? DienThoai { get; set; }
    public bool TrangThai { get; set; } = true;
    public int? NguoiDungID { get; set; }
    public DateTime TaoLuc { get; set; }
    public DateTime? CapNhatLuc { get; set; }
    [JsonIgnore] public NguoiDung? NguoiDung { get; set; }
    [JsonIgnore] public ICollection<DangKyThi> DangKyThiRecords { get; set; } = [];
}
